Sources/Logging/LoggerType.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  LoggerType.swift
+//
+//  Created by Nacho Soto on 11/29/22.
+
+import Foundation
+
+/// A type that can receive logs of different levels.
+protocol LoggerType {
```

Comment:
this is a pattern that I wish we had used a lot more in the codebase, the inherit + replace approach feels more cumbersome and dangerous to me